src/vehicle_move.cpp

```diff
@@ -1595,6 +1595,56 @@ vehicle *vehicle::act_on_map()
     return g->m.move_vehicle( *this, dp, mdir );
 }
 
+bool vehicle::level_vehicle()
+{
+    if( !g->m.has_zlevels() || ( is_flying && is_rotorcraft() ) ) {
+        return true;
+    }
+    // make sure that all parts are either supported across levels or on the same level
+    std::map<int, bool> no_support;
+    for( vehicle_part &prt : parts ) {
+        if( prt.info().location != part_location_structure ) {
+            continue;
+        }
+        const tripoint part_pos = global_part_pos3( prt );
+        if( no_support.find( part_pos.z ) == no_support.end() ) {
+            no_support[part_pos.z] = part_pos.z > -OVERMAP_DEPTH;
+        }
+        if( no_support[part_pos.z] ) {
+            no_support[part_pos.z] &= g->m.has_flag_ter_or_furn( TFLAG_NO_FLOOR, part_pos ) &&
```

Comment:
```suggestion
            no_support[part_pos.z] = g->m.has_flag_ter_or_furn( TFLAG_NO_FLOOR, part_pos ) &&
```
You have already checked that `no_support[part_pos.z]`, so this `... = true && ...`

Reply:
I'm copying the logic from `check_falling_or_floating()` below.

Reply:
Fixed.